pkg/builder/csolution/builder.go

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2023 Arm Limited. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package csolution
+
+import (
+	builder "cbuild/pkg/builder"
+	"cbuild/pkg/builder/cproject"
+	utils "cbuild/pkg/utils"
+	"errors"
+	"os"
+	"path/filepath"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type CSolutionBuilder struct {
+	builder.BuilderParams
+}
+
+func (b CSolutionBuilder) listContexts(quite bool) (contexts []string, err error) {
+	args := []string{"list", "contexts", "--solution=" + b.InputFile}
+
+	if b.Options.Filter != "" {
+		args = append(args, "--filter="+b.Options.Filter)
+	}
+
+	if !b.Options.Schema {
+		args = append(args, "--no-check-schema")
+	}
+
+	csolutionBin := filepath.Join(b.InstallConfigs.BinPath, "csolution"+b.InstallConfigs.BinExtn)
+	if _, err := os.Stat(csolutionBin); os.IsNotExist(err) {
+		log.Error("csolution was not found")
+		return nil, err
+	}
+
+	output, err := b.Runner.ExecuteCommand(csolutionBin, quite, args...)
+	if err != nil {
+		log.Error("error executing 'cbuild list-contexts'")
+		return nil, err
+	}
+	output = strings.ReplaceAll(output, " ", "")
+	if output != "" {
+		contexts = strings.Split(strings.ReplaceAll(strings.TrimSpace(output), "\r\n", "\n"), "\n")
+	}
+	return contexts, nil
+}
+
+func (b CSolutionBuilder) listToolchains(quite bool) (toolchains []string, err error) {
+	csolutionBin := filepath.Join(b.InstallConfigs.BinPath, "csolution"+b.InstallConfigs.BinExtn)
+	if _, err := os.Stat(csolutionBin); os.IsNotExist(err) {
+		log.Error("csolution was not found")
+		return toolchains, err
+	}
+
+	args := []string{"list", "toolchains", "--solution=" + b.InputFile}
+
+	output, err := b.Runner.ExecuteCommand(csolutionBin, quite, args...)
+	if err != nil {
+		log.Error("error executing 'cbuild list-toolchains'")
```

Comment:
```suggestion
		log.Error("error executing 'cbuild list toolchains'")
```

Reply:
Addressed in one of the changes done in [Improvements](https://github.com/Open-CMSIS-Pack/cbuild/pull/29/commits/472704065a8867715b9fefc2c651f85f036e9c02)